SwrveSDK/src/main/java/com/swrve/sdk/SwrveAssetsManager.java

```diff
@@ -0,0 +1,69 @@
+package com.swrve.sdk;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+
+public class SwrveAssetsManager {
+    private final String LOG_TAG = "SwrveAssetsManager";
+    private final String cdnRoot;
+    protected Set<String> assetsOnDisk;
+    private Set<String> failedAssets;
+    protected boolean assetsCurrentlyDownloading;
+    protected File imageCacheDir;
+
+
+    public SwrveAssetsManager(File imageCacheDir, String cdnRoot) {
+        this.imageCacheDir = imageCacheDir;
+        this.cdnRoot = cdnRoot;
+        this.assetsOnDisk = new HashSet<String>();
+        this.failedAssets = new HashSet<String>();
+        this.assetsCurrentlyDownloading = false;
+    }
+
+    public String getCDNUrlForAssetPath(String assetPath) {
+        return cdnRoot + assetPath;
+    }
+
+    public boolean isAssetsCurrentlyDownloading() {
+        return assetsCurrentlyDownloading;
+    }
+
+    public void setAssetsCurrentlyDownloading(boolean assetsCurrentlyDownloading) {
+        this.assetsCurrentlyDownloading = assetsCurrentlyDownloading;
+    }
+
+    public void addAsset(String asset) {
+        failedAssets.remove(asset);
+        if (containsAsset(asset)) {
+            SwrveLogger.w(LOG_TAG, "Asset already present in SwrveAssetManager");
```

Comment:
I wasn't sure. In theory this should never actually happen so It seems like a warning. But sure I'll retag it to I once dev kicks off again on this branch